File: InsuraTech.Domain/Events/ClaimRegisteredEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InsuraTech.Domain.Common;

namespace InsuraTech.Domain.Events
{
    public sealed class ClaimRegisteredEvent : IDomainEvent
    {
        public Guid EventId { get; } = Guid.NewGuid();
        public Guid Id => EventId;

        public DateTime OccurredOn { get; } = DateTime.UtcNow;
        public string EventType => nameof(ClaimRegisteredEvent);

        public Guid ClaimId { get; }
        public Guid PolicyId {  get; }
        public string PolicyNumer {  get; }
        public Decimal ClaimAmount {  get; }

        public ClaimRegisteredEvent(Guid claimId, Guid policyId, string policyNumer, decimal claimAmount)
        {
            ClaimId = claimId;
            PolicyId = policyId;
            PolicyNumer = policyNumer;
            ClaimAmount = claimAmount;
        }
    }
}

File: backend_v2/src/main/java/com/insuratech/domain/events/ClaimRegisteredEvent.java
package com.insuratech.domain.events;

import com.insuratech.domain.common.IDomainEvent;

import java.time.Instant;

public record ClaimRegisteredEvent(String claimId, String policyId, String claimType, Instant occurredOn) implements IDomainEvent {
    public ClaimRegisteredEvent(String claimId, String policyId, String claimType) {
        this(claimId, policyId, claimType, Instant.now());
    }
}
